anta/reporter/csv_reporter.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+"""CSV Report management for ANTA."""
+
+# pylint: disable = too-few-public-methods
+from __future__ import annotations
+
+import csv
+import logging
+import pathlib
+from typing import TYPE_CHECKING
+
+from pydantic import BaseModel
+
+if TYPE_CHECKING:
+    from anta.result_manager import ResultManager
+    from anta.result_manager.models import TestResult
+
+logger = logging.getLogger(__name__)
+
+
+class ReportCsv:
+    """Build a CSV report."""
+
+    class Headers(BaseModel):
```

Comment:
Yeah I think a dataclass would be cleaner. We can put it in a models.py in either reporter or result_manager package and we can reuse it.

Reply:
(I would reuse in the MD report since they are the same)

Reply:
Convert all class Headers to dataclass. But still per report type since there is no common headers across all reports